### PR TITLE
Add: Range , Events Search, Shop API, Fix:API-Doc

### PR DESCRIPTION
**Pull request by 123r123**

### CODE REVIEW

File: API-Doc/README.md
Comment (i-am-harveyt) [inline API-Doc/README.md]: I think it is better to remain latitude and longitude.
Because some abbreviation might not be trivial enough.

File: api/1.0/controller/events/create.controller.js
Comment (i-am-harveyt) [inline api/1.0/controller/events/create.controller.js]: I'll handle this part on my branch, you can restore this change to avoid conflict

File: api/1.0/controller/events/rangeShop.controller.js
Comment (i-am-harveyt) [inline api/1.0/controller/events/rangeShop.controller.js]: I think `!(lat && lon && latitude && longitude)` may perform a little bit better, since `||` make it necessary to check both sides.

File: api/1.0/controller/users/verifyToken.controller.js
Comment (i-am-harveyt) [inline api/1.0/controller/users/verifyToken.controller.js:1]: It seems like a general-purpose utility function, it may be placed in `api/1.0/utils/jwt/` and renamed as `verifyToken.util.js`.

File: api/1.0/model/events/create.model.js
Comment (i-am-harveyt) [inline api/1.0/model/events/create.model.js:1]: I'll handle this in my branch either, you may just restore changes to avoid a conflict

File: api/1.0/model/events/range.model.js
Comment (i-am-harveyt) [inline api/1.0/model/events/range.model.js:1]: `ST_Distance_Sphere` may not give us a Manhattan distance, the latter one is a more practical distance calculation method, you may want to give it a try

File: api/1.0/model/events/range.model.js
Comment (i-am-harveyt) [inline api/1.0/model/events/range.model.js]: I recommend rename `param` as `params`, since there are more than 1 parameters.

File: api/1.0/model/events/rangeShop.model.js
Comment (i-am-harveyt) [inline api/1.0/model/events/rangeShop.model.js]: Same as the comment in `range.model.js`.

File: api/1.0/model/events/search.model.js
Comment (i-am-harveyt) [inline api/1.0/model/events/search.model.js:6]: Same as the comment written in `range.model.js`.

File: mysql/JoinEat.sql
Comment (i-am-harveyt) [inline mysql/JoinEat.sql:1]: Great Job!

**Review by i-am-harveyt — COMMENTED** [review]
There are some comments left in files, you may want to take a look on it.
Since some of them may leads to dramatic changes, I may not approve this now.

File: api/1.0/model/events/range.model.js
Comment (123r123) [reply api/1.0/model/events/range.model.js:1]: "曼哈頓距離只適用於平面上的直線距離計算，不考慮任何曲線形狀，也不適用於地理空間數據。如果你處理的是地理坐標或需要考慮地球的曲率，則應該使用 ST_Distance_Sphere 或其他適用的地理空間函數。"  from ChatGPT

File: api/1.0/model/events/create.model.js
Comment (123r123) [reply api/1.0/model/events/create.model.js:1]: This will affect the operation, so I'll keep same.

File: api/1.0/controller/events/range.controller.js
Comment (i-am-harveyt) [inline api/1.0/controller/events/range.controller.js]: The file may be named as `transfromTimeFormat.util.js`

File: api/1.0/controller/events/search.controller.js
Comment (i-am-harveyt) [inline api/1.0/controller/events/search.controller.js]: Same concept as the comment left in `range.controller.js`

**Review by i-am-harveyt — COMMENTED** [review]
Nice job.
There are some file name issue, after fixing them, this will be able to merge.

**Review by i-am-harveyt — APPROVED** [review]
LGTM

**Pull request merged**
